Add section about the acronym rule to the style guide contrib (#4415)
umbraco/UmbracoDocs: Contribute/Style-Guide/index.md

The list of Umbraco Terms includes, but is not limited to *Umbraco*, *backoffice*, *Document Type* and *Umbraco Forms*.

### Defined acronyms

> All first-time uses of an acronym in the documentation needs to be accompanied by a definition of that acronym. If an acronym is nto defined on its first use in an article, the checker will with a warning.

Reviewer: ```suggestion
All first-time uses of an acronym in the documentation need to be accompanied by a definition of that acronym. If an acronym is not defined on its first use in an article, the checker will give a warning.
```
